File: Algorithm/9_PalindromeNumber/main.py
from math import ceil, floor


def isPalindrome(self, x: int) -> bool:
    if x < 0:
        return False
    num = str(x)
    r = 0
    l = 0
    if len(num) % 2 > 0:
        med = int(ceil(len(num)/2))
        r, l = med-1, med-1
    else:
        med = int(len(num)/2)
        r, l = med - 1, med
    error = False
    while l >= 0 and r < len(num):
        print(num[l], num[r])
        if num[r] != num[l]:
            error = True
            break
        r += 1
        l -= 1
    return error == False


print(isPalindrome(10, 100))

File: Algorithm/9_PalindromeNumber/main.cpp
#include <bits/stdc++.h>
using namespace std;
bool isPalindrome(int x)
{
    if (x < 0)
    {
        return false;
    }
    string num = to_string(x);
    int r, l;
    if (num.size() % 2 > 0)
    {
        l = (int)(num.size() / 2);
        r = (int)(num.size() / 2);
    }
    else
    {
        l = (int)(num.size() / 2) - 1;
        r = (int)(num.size() / 2);
    }
    while (l >= 0 && r < num.size())
    {
        if (num[l] != num[r])
            return false;
        l++;
        r--;
    }
    return true;
}
int main()
{
    cout << isPalindrome(-10011) << endl;
    return 0;
}